daemon/src/routes_taker.rs

```diff
@@ -159,25 +160,48 @@ pub async fn post_cfd_action(
 }
 
 #[derive(Debug, Clone, Deserialize)]
-pub struct WalletReinitialiseRequest {
-    pub seed_words: String,
+pub struct WalletRestoreRequest {
+    pub mnemonic: Mnemonic,
 }
 
-#[rocket::post("/wallet/reinitialise", data = "<wallet_reinitialise_request>")]
-pub async fn post_wallet_reinitialise(
-    wallet_reinitialise_request: Json<WalletReinitialiseRequest>,
+#[rocket::post("/wallet/restore", data = "<wallet_restore_request>")]
+pub async fn post_wallet_restore(
+    wallet_restore_request: Json<WalletRestoreRequest>,
     taker: &State<Taker>,
-) -> Result<status::Accepted<()>, HttpApiProblem> {
+) -> Result<(), HttpApiProblem> {
     taker
-        .reinitialise_wallet(&wallet_reinitialise_request.seed_words)
+        .restore_wallet(wallet_restore_request.mnemonic.clone())
         .await
         .map_err(|e| {
             HttpApiProblem::new(StatusCode::INTERNAL_SERVER_ERROR)
-                .title("Wallet recover request failed")
+                .title("Wallet restore request failed")
                 .detail(e.to_string())
         })?;
 
-    Ok(status::Accepted(None))
+    Ok(())
+}
+
+#[rocket::get("/wallet/mnemonic")]
+pub async fn get_wallet_mnemonic(taker: &State<Taker>) -> Result<Json<Mnemonic>, HttpApiProblem> {
+    let mnemonic = taker.backup_wallet().await.map_err(|e| {
+        HttpApiProblem::new(StatusCode::INTERNAL_SERVER_ERROR)
+            .title("Wallet mnemonic request failed")
+            .detail(e.to_string())
+    })?;
+
+    Ok(Json::from(mnemonic))
```

Comment:
Have you verified what this actually serializes to? I am unsure.

Reply:
`Mnemonic` serializes to a space separated world list but I have no idea what rocket + wrapping it `Json` does.

Reply:
Can we verify with a test that it actually serializes to JSON that we want to handle on the frontend?